### ISSUE TITLE
Extraneous characters displayed in zsh output

### ISSUE THREAD

**Issue by monguin61** [michaeldfallen/git-prompt#68]
I'm brand new to zsh, so this may be my mistake. The zsh examples given in the README and the usage message don't work for me, so I experimented and ended up using this:

```
%$(git-radar --zsh --fetch)
```

which results in this:

```
%{ git:(𝘮 34 ⇄ 65 feature/AUD-1140) 2A
```

note the '%{' at the beginning, which seems to be caused by the git-radar command, not by some other part of the prompt variable (this is a minimal working example). For what it's worth, within a longer PROMPT variable, the %{ is colored according to the preceding color escape codes.

This is on OS X Yosemite, in iTerm.

**Comment by michaeldfallen** [comment]
Can you post your `export PROMPT` line from the `~/.zshrc`? I think what's happening is the `%` you've put at the front is escaping the first character in the non-printing character boundary, which is `%{`, so you get `%{` output from `%%{`. That's my guess.

If I could see your `.zshrc` I can have a better look at it.

**Comment by michaeldfallen** [comment]
From [ZSH Docs - Prompt Expansion](http://zsh.sourceforge.net/Doc/Release/Prompt-Expansion.html):

> If the PROMPT_PERCENT option is set, certain escape sequences that start with ‘%’ are expanded. Many escapes are followed by a single character, although some of these take an optional integer argument that should appear between the ‘%’ and the next character of the sequence. More complicated escape sequences are available to provide conditional expansion.

**Comment by monguin61** [comment]
Below are the several different prompts I've tried. There is nothing else in my .zshrc, I just started using ZSH and haven't copied anything else over from other rc files yet. It looks like you're right about the '%' escaping, maybe if I read through more of the ZSH docs it will become clear to me what's going on. 

```
# my basic BASH prompt, no color
#PS1="[%D %T] %~
#alan@MBP (%j/%!)$
# result:
# [15-10-21 10:56:26] ~/src
# alan@MBP (0/30)$

# my basic BASH prompt, with color
#export PROMPT="%F{red}[%D %*] %~
#%F{red}alan@MBP (%F{cyan}%j%F{red}/%!)$%f "
# result:
# [15-10-21 10:56:26] ~/src
# alan@MBP (0/30)$

# my basic BASH prompt, with color and git-radar
#export PROMPT="%F{red}[%D %*] %~%f %$(git-radar --zsh --fetch)
#%F{red}alan@MBP (%F{cyan}%j%F{red}/%!)$%f "
# result:
# [15-10-21 10:56:05] ~/src %{ git:(𝘮 402 ⇄ 8 feature/PLAT-187 1↑) 10M
# alan@MBP (0/29)$
```

This is the best thing I've found so far:

```
# other attempts
#export PROMPT="%$(git-radar --zsh --fetch) $ "
# result:
# %{ git:(𝘮 402 ⇄ 8 feature/PLAT-187 1↑) 10M $
```

And this is your suggestion from issue 46, which isn't working for me:

```
export PROMPT=$'$(git-radar) '
# result:
# $(git-radar)

#export PROMPT=$'$(git-radar --zsh --fetch) '
# result:
# $(git-radar --zsh --fetch)
```

Thanks for your help. I thought there might be a simple fix, visible to someone who has used ZSH more - if that's not the case, don't worry about it.

**Comment by michaeldfallen** [comment]
Ok, I've got it figured out.

You've only just started with ZSH so likely you haven't actually enabled prompt expansion. It's not something ZSH does by default but most people turn it on.

To do that add this to `~/.zshrc`:

``` bash
setopt PROMPT_SUBST
```

Now use the form I suggested for your prompt:

``` bash
export PROMPT=$'$(git-radar --zsh --fetch) '
```

I have a set of sane defaults I use at [dotfiles/zsh/config.zsh](https://github.com/michaeldfallen/dotfiles/blob/master/zsh/config.zsh). Most of them are commented but YMMV for which you like to use. `zstyle ':completion:*' menu select` is particularly nice as it lets you arrow key move through completions, making auto-complete feel so much more modern.

Welcome to ZSH!
